Point Cloud refactoring

gchoqueux opened the pull request:
## Description
 * introduce `PointCloudSource`, this class have the specific parser and fetcher.
 * introduce `PointCloudNode`, this class structures, the data, by nodes that make up the `octree`. `PointCloudNode` is to have its own loading methods. 


## Motivation and Context
* It's the continuation of the refactoring starting with #1288 

## The next steps
* it's clean the process methods in `PointCloudLayer` and move `BoxHelper` in tools.
* remove `lopocs` support.

## BREAKING CHANGES
To add points cloud, you need to instance:
* `PointCloudSource` to specify parameters to fetch data.
* `PointCloudLayer` to add this points cloud in viewer.

gchoqueux committed:
refactor(pointCloud): use BoxHelper to display point cloud boudingBox.

mbredif commented:
The generic names used in this PR do not make explicit the fact that the point clouds are indeed served using the potree format (hrc index and potree leaves...).
All classes tied to Potree should have Potree in their name I think.

gchoqueux commented:
Yes, but for the moment itowns supports `lopocs`. I would like remove it and rename all class to `PotreeXXX`.

zarov commented on src/Layer/PointCloudLayer.js:
```diff
@@ -284,41 +87,29 @@ class PointCloudLayer extends GeometryLayer {
-        if (!config.file) {
-            throw new Error('New PointCloudLayer: file is required');
-        }
         super(id, new THREE.Group());
```
You can pass the `config` object and avoid adding the next line (and maybe some others)

gchoqueux replied:
ok

zarov commented on src/Source/PointCloudSource.js:
```diff
@@ -0,0 +1,65 @@
+        this.fetcher = Fetcher.arrayBuffer;
+        this.extensionOctree = 'hrc';
+
+        this.whenReady = (source.metadata ? Promise.resolve(source.metadata) : Fetcher.json(`${this.url}/${this.file}`, this.networkOptions))
```
Please add some documentation for `metadata`.

gchoqueux replied:
ok

zarov commented on test/unit/pointcloudlayerparsing.js:
```diff
@@ -1,51 +1,93 @@
-        const layer = {
-            material: { defines: {} },
-        };
+const renderer = new Renderer();
```
Please move those variables (`renderer`,  `placement`...) between the `describe` and `it` section, so it stays in this scope.

gchoqueux replied:
ok

zarov reviewed (commented):
LGTM, please see my comments and we are good to merge

gchoqueux committed:
refactor(PointCloud): change tests units.

zarov commented on src/Source/PointCloudSource.js:
```diff
@@ -0,0 +1,67 @@
+        this.extensionOctree = 'hrc';
+
+        // For cloud specification visit:
+        // https://github.com/PropellerAero/potree-propeller-private/blob/master/docs/file_format.md#cloudjs
```
Please add a quick copy of this in the documentation of the `constructor` above, in case this page disappears, and to ease the process to the user

gchoqueux replied:
done

gchoqueux committed:
refactor(PointCloud): add PointCloudSource and PointCloudNode.

BREAKING CHANGE:
To add points cloud, you need to instance:
- PointCloudSource to specify parameters to fetch data.
- PointCloudLayer to add this points cloud in viewer.
